lib/mongo/collection/view/iterable.rb

```diff
@@ -37,14 +37,18 @@ module Iterable
         def each
           @cursor = nil
           session = client.send(:get_session, @options)
-          @cursor = if respond_to?(:write?, true) && write?
-            server = server_selector.select_server(cluster, nil, session)
-            result = send_initial_query(server, session)
-            Cursor.new(view, result, server, session: session)
-          else
-            read_with_retry_cursor(session, server_selector, view) do |server|
-              send_initial_query(server, session)
+          if QueryCache.enabled?
+            unless @cursor = cached_cursor
+              @cursor = select_cursor(session)
+              QueryCache.cache_table[cache_key] = @cursor
+            end
+            if limit
+              @cursor.limit_docs = limit
+            else
+              @cursor.limit_docs = nil
```

Comment:
If my other comment about the `limit_docs` method is correct, you should also be able to remove this conditional statement.